### PR TITLE
Set full flow

### PR DESCRIPTION
**Pull request by kevinch**
#### What is the purpose of this pull request?
Set full flow of Credit Control payment at Checkout.

Loading state:
screen shot 2018-06-22 at 14 15 14

Account not found:
screen shot 2018-06-22 at 13 35 35

Account found, not enough credit:
screen shot 2018-06-22 at 14 14 11

Account found, credit ok:
screen shot 2018-06-22 at 14 14 27

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: extension_point_checkout/react/components/CheckoutListener.js [hunk 1]
```diff
@@ -7,57 +7,87 @@ import { path } from 'ramda'
...
-      window.vtexjs.checkout.orderForm.clientProfileData.email
+      path(
+        ['vtexjs', 'checkout', 'orderForm', 'clientProfileData', 'email'],
+        window,
+      )
     ) {
-      clientEmail = window.vtexjs.checkout.orderForm.clientProfileData.email
-      this.fetchData(clientEmail)
+      const orderForm = window.vtexjs.checkout.orderForm
+      const paymentValue = orderForm.value / 100
```
Comment (klzns) [inline extension_point_checkout/react/components/CheckoutListener.js]: O valor da compra muda e essa verificação tem que ser feita a cada mudança no valor do carrinho e no email.

File: extension_point_checkout/react/components/EnoughCredit.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                            }): `}
+                        {intl.formatNumber(i.value / 100, {
+                          style: 'currency',
+                          currency: currencyCode,
+                        })}
+                      </option>
+                    ) : (
+                      <option key={i.number} value={i.number}>
+                        {i.number}
+                        {intl.formatMessage({ id: 'xOf' })}{' '}
```
Comment (klzns) [inline extension_point_checkout/react/components/EnoughCredit.js]: Nao traduzir assim, usar interpolacao do i18n.

Exemplo:

```js
  "singleInstallment": "Pagamento à vista - {{- value}}",
  "installments": "{{installments}}x de {{- value}} sem juros",
  "installmentsUnknownInterest": "{{installments}} parcelas",
  "installmentsWithInterest": "{{installments}}x de {{- value}} com juros de {{interest}}% a.m.",
```

```js
  "singleInstallment": "Total - {{- value}}",
  "installments": "{{installments}} mensualités de {{- value}} (sans intérêt)",
  "installmentsUnknownInterest": "{{installments}} mensualités",
  "installmentsWithInterest": "{{installments}} mensualités de {{- value}} (taux d'intérêt mensuel de {{interest}}%)",
```

Note que em frances nao usamos o "x"

File: extension_point_checkout/react/index.js [hunk 3]
```diff
@@ -1,35 +1,78 @@
...
+
+            if (dataVerified) {
+              return (
+                <div>
+                  {!hasAccount &&
+                    clientEmail && (
+                      <AccountNotFound clientEmail={clientEmail} />
+                    )}
+
+                  {hasEnougthCredit
```
Comment (klzns) [inline extension_point_checkout/react/index.js]: Typo "enough"

File: extension_point_checkout/react/components/CheckoutListener.js (hunk 1 above)
Comment (kevinch) [reply extension_point_checkout/react/components/CheckoutListener.js]: @klzns sabe explicar melhor esse ponto?

**Pull request merged**